packages/server/src/automations/triggers.ts

```diff
@@ -156,14 +149,26 @@ export function isAutomationResults(
   )
 }
 
+interface AutomationTriggerParams {
+  fields: Record<string, any>
+  timeout?: number
+  appId?: string
+  user?: UserBindings
+}
+
+export async function externalTrigger(
+  automation: Automation,
+  params: AutomationTriggerParams,
+  options: { getResponses: true }
+): Promise<AutomationResults | DidNotTriggerResponse>
+export async function externalTrigger(
+  automation: Automation,
+  params: AutomationTriggerParams,
+  options?: { getResponses: false }
+): Promise<AutomationJob | DidNotTriggerResponse>
```

Comment:
This function either queues an `AutomationJob`, or runs an automation synchronously and returns an `AutomationResults` based on what the value of `options.getResponses` is. These overloads narrow down these types to help the testing endpoint know it has hold of an `AutomationResults` object, never actually has to deal with the `AutomationJob` type.